src/slurm_plugin/slurm_resources/instance_manager.py

```diff
@@ -0,0 +1,347 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
The reason not putting instance_manager with slurm_resources is that it will cause circular dependency, it both depend on slurm_resources.py and slurm_commands.py By moving it out, we can solve this problem. Besides, instance manager is a large class about 350 lines's of codes, I think we can separate it to a different file to not making the sluirm_resources.py too long.